runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/AwsSigner.kt

```diff
@@ -31,4 +31,17 @@ public interface AwsSigner {
         prevSignature: ByteArray,
         config: AwsSigningConfig,
     ): AwsSigningResult<Unit>
+
+    /**
+     * Signs a chunked payload's trailer according to the supplied signing configuration
+     * @param trailingHeaders The canonicalized trailing headers
+     * @param finalChunkSignature The signature of the final payload chunk
+     * @param config The signing configuration
+     * @return The signing result, which should be appended as a trailing header itself, named `x-amz-trailer-signature`
+     */
+    public suspend fun signChunkTrailer(
+        trailingHeaders: ByteArray,
```

Comment:
Question/correctness: Should this be required to be the canonicalized version of the headers? We don't force that anywhere else in signing and it seems like an internal concern of signing. Should we be taking a `Headers` instance instead and leave canonicalization to the implementation(s)?

Reply:
We could leave canonicalizing the headers to the implementations, but there are already some cases of [receiving canonicalized requests in the signer](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/auth/aws-signing-default/common/src/aws/smithy/kotlin/runtime/auth/awssigning/SignatureCalculator.kt#L52), so I was following that.

I figured whenever the request gets canonicalized, it could also canonicalize the trailing headers to prevent duplicating the canonicalization code.

I think it would be better to [reuse that code](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/auth/aws-signing-default/common/src/aws/smithy/kotlin/runtime/auth/awssigning/Canonicalizer.kt#L105) rather than rewriting it in each `signChunkTrailer` implementation.

Reply:
I ended up adding canonicalization code where it's needed rather than assuming the input is canonicalized. All we need to do for canonicalization is sort alphabetically and lowercase the header names